src/webview/java/org/medicmobile/webapp/mobile/EmbeddedBrowserActivity.java

```diff
@@ -435,6 +451,26 @@ public void onReceivedError(WebView view, WebResourceRequest request, WebResourc
 							"}", error.getErrorCode(), error.getDescription()), false);
 				}
 			}
+
+			@Override
+			public void onPageStarted(WebView view, String url, Bitmap favicon) {
+				trace(this, "onPageStarted() :: url: %s, isMigrationRunning: %s", url, isMigrationRunning);
+				if (isMigrationRunning && url.contains("/login")) {
+					isMigrationRunning = false;
+					CookieManager cookieManager = CookieManager.getInstance();
+					String cookie = cookieManager.getCookie(appUrl);
+					if (cookie == null) {
```

Comment:
What happens if a user's cookies have expired (or be wiped manually)? I think it'll work... migration will pick up the IndexedDB, then on restart `isMigrationRunning` will be false, and the user will go to the login page (as expected). Is that right? Have you tested this case?

Reply:
Yes, I tried to install this version over a XWalk installation that has loaded the webapp but logged out, and the migration runs and then the app display the login page as expected.

Also installed the app without previous installations of Xwalk to ensure the migration process is not started. Same if it was a XWalk installation but with no data previously loaded.